Static code analysis errors with numpy 2.2.0

The "Static code analysis" action currently fails with several mypy errors, most probably due to the release of [numpy 2.2](https://numpy.org/devdocs/release/2.2.0-notes.html) yesterday, which contains "Many improved annotations."

For example https://github.com/phasorpy/phasorpy/actions/runs/12243727085/job/34153909334:

```
Run python -m mypy
src/phasorpy/color.py:63: error: Incompatible return value type (got "tuple[Any | float | list[float] | list[list[Any]], Any | float | list[float] | list[list[Any]], Any | float | list[float] | list[list[Any]]]", expected "tuple[float, float, float] | ndarray[tuple[int, ...], dtype[Any]]")  [return-value]
src/phasorpy/color.py:63: error: Value of type "float | list[float] | list[list[float]] | list[list[list[Any]]]" is not indexable  [index]
src/phasorpy/utils.py:184: error: Incompatible types in assignment (expression has type "ndarray[tuple[int, ...], dtype[Any]]", variable has type "ndarray[tuple[int, int], dtype[Any]]")  [assignment]
src/phasorpy/utils.py:186: error: Incompatible types in assignment (expression has type "ndarray[tuple[int, ...], dtype[Any]]", variable has type "ndarray[tuple[int, int], dtype[Any]]")  [assignment]
src/phasorpy/phasor.py:703: error: Incompatible types in assignment (expression has type "ndarray[tuple[int, ...], dtype[float64]]", variable has type "ndarray[tuple[int], dtype[float64]]")  [assignment]
src/phasorpy/phasor.py:2133: error: Incompatible types in assignment (expression has type "ndarray[tuple[int, ...], dtype[float64]]", variable has type "ndarray[tuple[int, int, int], dtype[float64]]")  [assignment]
src/phasorpy/plot.py:340: error: No overload variant of "__setitem__" of "ndarray" matches argument types "Any", "None"  [call-overload]
src/phasorpy/plot.py:340: note: Possible overload variants:
src/phasorpy/plot.py:340: note:     def __setitem__(self, str | list[str], Buffer | _SupportsArray[dtype[Any]] | _NestedSequence[_SupportsArray[dtype[Any]]] | bool | int | float | complex | str | bytes | _NestedSequence[bool | int | float | complex | str | bytes], /) -> None
src/phasorpy/plot.py:340: note:     def __setitem__(self, SupportsIndex | slice | EllipsisType | _SupportsArray[dtype[numpy.bool[builtins.bool]] | dtype[integer[Any]]] | _NestedSequence[_SupportsArray[dtype[numpy.bool[builtins.bool]] | dtype[integer[Any]]]] | builtins.bool | int | _NestedSequence[builtins.bool | int] | tuple[SupportsIndex | slice | EllipsisType | _SupportsArray[dtype[numpy.bool[builtins.bool]] | dtype[integer[Any]]] | _NestedSequence[_SupportsArray[dtype[numpy.bool[builtins.bool]] | dtype[integer[Any]]]] | builtins.bool | int | _NestedSequence[builtins.bool | int] | None, ...] | None, Buffer | _SupportsArray[dtype[Any]] | _NestedSequence[_SupportsArray[dtype[Any]]] | bool | int | float | complex | str | bytes | _NestedSequence[bool | int | float | complex | str | bytes], /) -> None
src/phasorpy/plot.py:342: error: No overload variant of "__setitem__" of "ndarray" matches argument types "Any", "None"  [call-overload]
src/phasorpy/plot.py:342: note: Possible overload variants:
src/phasorpy/plot.py:342: note:     def __setitem__(self, str | list[str], Buffer | _SupportsArray[dtype[Any]] | _NestedSequence[_SupportsArray[dtype[Any]]] | bool | int | float | complex | str | bytes | _NestedSequence[bool | int | float | complex | str | bytes], /) -> None
src/phasorpy/plot.py:342: note:     def __setitem__(self, SupportsIndex | slice | EllipsisType | _SupportsArray[dtype[numpy.bool[builtins.bool]] | dtype[integer[Any]]] | _NestedSequence[_SupportsArray[dtype[numpy.bool[builtins.bool]] | dtype[integer[Any]]]] | builtins.bool | int | _NestedSequence[builtins.bool | int] | tuple[SupportsIndex | slice | EllipsisType | _SupportsArray[dtype[numpy.bool[builtins.bool]] | dtype[integer[Any]]] | _NestedSequence[_SupportsArray[dtype[numpy.bool[builtins.bool]] | dtype[integer[Any]]]] | builtins.bool | int | _NestedSequence[builtins.bool | int] | None, ...] | None, Buffer | _SupportsArray[dtype[Any]] | _NestedSequence[_SupportsArray[dtype[Any]]] | bool | int | float | complex | str | bytes | _NestedSequence[bool | int | float | complex | str | bytes], /) -> None
Found 8 errors in 4 files (checked 15 source files)
```

--- a/.github/workflows/run-tests.yml
+++ b/.github/workflows/run-tests.yml
@@ -144,7 +144,7 @@ jobs:
           python -m pip install --upgrade pip
           python -m pip install --editable .
           python -m pip install -r requirements_dev.txt
-          python -m pip install -r requirements_min.txt
+          python -m pip install -U "numpy<2.2.0"
       - name: Test with black
         run: |
           python -m black --check src/phasorpy tutorials docs

